fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVJobManager.java

```diff
@@ -470,56 +317,75 @@ public void replayDropJobs(List<Long> jobIds) {
     }
 
     public void replayUpdateJob(ChangeMTMVJob changeJob) {
-        updateJob(changeJob, true);
+        MTMVJob mtmvJob = idToJobMap.get(changeJob.getJobId());
+        if (mtmvJob != null) {
+            mtmvJob.updateJob(changeJob, true);
+        }
     }
 
     public void replayCreateJobTask(MTMVTask task) {
         taskManager.replayCreateJobTask(task);
     }
 
     public void replayDropJobTasks(List<String> taskIds) {
-        taskManager.dropTasks(taskIds, true);
+        taskManager.dropHistoryTasks(taskIds, true);
     }
 
     public void removeExpiredJobs() {
```

Comment:
change to private.
And check other method, if it only be used inside this class, change to private